src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -245,46 +246,22 @@ private async Task<bool> WriteResultsAsync(TContext rootContext)
 
                         while (context?.AnalysisComplete == true)
                         {
-                            var cachingLogger = (CachingLogger)context.Logger;
-                            IDictionary<ReportingDescriptor, IList<Result>> results = cachingLogger.Results;
-
-                            if (results?.Count > 0)
+                            if (_computeHashes)
```

Comment:
>_computeHashes [](http://example.com/codeflow?start=32&length=14)

are you merging the concepts of computing hashes and caching log results? is this what we want to do?

Reply:
Before my change, the results were not cached. each file had its own logger.

With my change, if hash is enabled, we would re-use the logger from the first hash. Otherwise, everything is the same.

This is mimicking the single thread way but using cachingLogger